testsuite/forge/src/main.rs

```diff
@@ -143,6 +144,29 @@ impl PublicUsageTest for TransferCoins {
     }
 }
 
+#[derive(Debug)]
+struct RestartValidator;
+
+impl Test for RestartValidator {
+    fn name(&self) -> &'static str {
+        "restart_validator"
+    }
+}
+
+impl NetworkTest for RestartValidator {
+    fn run<'t>(&self, ctx: &mut NetworkContext<'t>) -> Result<()> {
+        let node_id = NodeId::new(0);
+        let node = ctx.swarm().validator_mut(node_id);
+        node.health_check();
+        node.stop()?;
+        println!("Restarting node {}", node.node_id().as_inner());
+        node.start()?;
+        node.health_check();
```

Comment:
I think the health_check function is going to need to be tweaked in order to ensure that we are confident that the health check is successful since right now the result is just thrown away.

We should change the health_check function signature to be something like:
```
fn health_check(&mut self) -> Result<(), HealthCheckError>;

enum HealthCheckError {
Crashed,
NotStarted,
RpcFailure,
...
}

impl std::error::Error for HealthCheckError {}
```
Note that we should define a strict Error enum (and not reuse Anyhow for this)
And then ensure that the health_check implementation doesn't panic and instead returns an error on a failed healthcheck. This way we are forcing use to look at the result of the health check